src/test/crimson/test_messenger.cc

```diff
@@ -335,9 +345,11 @@ static seastar::future<> test_concurrent_dispatch(bool v2)
     return server->wait();
   }).finally([client] {
     logger().info("client shutdown...");
+    client->msgr->remove_dispatcher(*client);
     return client->msgr->shutdown();
   }).finally([server] {
     logger().info("server shutdown...");
+    server->msgr->remove_dispatcher(*server);
     return server->msgr->shutdown();
   }).finally([server, client] {
```

Comment:
is the captured client used here?

Reply:
I think it is captured here just to make sure those two objects not get destroyed before the continuation chain finishes:-)